src/fortran/rangen.fpp

```diff
@@ -1,380 +1,164 @@
 
-      SUBROUTINE RMMARD( RVEC,LENV,ISEQ )
+      external npygas
+      integer dummy
+      integer*8 bitgen
+      COMMON /npy/bitgen
 
-C-----------------------------------------------------------------------
-C  R(ANDO)M (NUMBER GENERATOR OF) MAR(SAGLIA TYPE) D(OUBLE PRECISION)
-C
-C  THESE ROUTINES (RMMARD,RMMAQD) ARE MODIFIED VERSIONS OF ROUTINES
-C  FROM THE CERN LIBRARIES. DESCRIPTION OF ALGORITHM SEE:
-C   http://wwwasdoc.web.cern.ch/wwwasdoc/cernlib.html (v113)
-C  IT HAS BEEN CHECKED THAT RESULTS ARE BIT-IDENTICAL WITH CERN
-C  DOUBLE PRECISION RANDOM NUMBER GENERATOR RMM48, DESCRIBED IN
-C   http://wwwasdoc.web.cern.ch/wwwasdoc/cernlib.html (v116)
-C  ARGUMENTS:
-C   RVEC   = DOUBLE PREC. VECTOR FIELD TO BE FILLED WITH RANDOM NUMBERS
-C   LENV   = LENGTH OF VECTOR (# OF RANDNUMBERS TO BE GENERATED)
-C   ISEQ   = # OF RANDOM SEQUENCE
-C
-C  VERSION OF D. HECK FOR DOUBLE PRECISION RANDOM NUMBERS.
-C-----------------------------------------------------------------------
+      call NPYGAS(gasdev, bitgen)
 
-      IMPLICIT NONE
+      end
 
-      INTEGER          KSEQ
-      PARAMETER        (KSEQ = 8)
-      COMMON /CRRANMA3/CD,CINT,CM,TWOM24,TWOM48,MODCNS
-      DOUBLE PRECISION CD,CINT,CM,TWOM24,TWOM48
-      INTEGER          MODCNS
+      REAL FUNCTION spgasdev( dummy )
+      implicit none
 
-      COMMON /CRRANMA4/C,U,IJKL,I97,J97,NTOT,NTOT2,JSEQ
-      DOUBLE PRECISION C(KSEQ),U(97,KSEQ),UNI
-      INTEGER          IJKL(KSEQ),I97(KSEQ),J97(KSEQ),
-     *                 NTOT(KSEQ),NTOT2(KSEQ),JSEQ
+      external npygas
+      integer dummy
+      integer*8 bitgen
+      COMMON /npy/bitgen
+      double precision rval
 
-      DOUBLE PRECISION RVEC(*)
-      INTEGER          ISEQ,IVEC,LENV
-      SAVE
-C-----------------------------------------------------------------------
+      call NPYGAS(rval, bitgen)
 
-      IF ( ISEQ .GT. 0  .AND.  ISEQ .LE. KSEQ ) JSEQ = ISEQ
-
-      DO  IVEC = 1, LENV
-        UNI = U(I97(JSEQ),JSEQ) - U(J97(JSEQ),JSEQ)
-        IF ( UNI .LT. 0.D0 ) UNI = UNI + 1.D0
-        U(I97(JSEQ),JSEQ) = UNI
-        I97(JSEQ)  = I97(JSEQ) - 1
-        IF ( I97(JSEQ) .EQ. 0 ) I97(JSEQ) = 97
-        J97(JSEQ)  = J97(JSEQ) - 1
-        IF ( J97(JSEQ) .EQ. 0 ) J97(JSEQ) = 97
-        C(JSEQ)    = C(JSEQ) - CD
-        IF ( C(JSEQ) .LT. 0.D0 ) C(JSEQ)  = C(JSEQ) + CM
-        UNI        = UNI - C(JSEQ)
-        IF ( UNI .LT. 0.D0 ) UNI = UNI + 1.D0
-C  AN EXACT ZERO HERE IS VERY UNLIKELY, BUT LET''S BE SAFE.
-        IF ( UNI .EQ. 0.D0 ) UNI = TWOM48
-        RVEC(IVEC) = UNI
-      ENDDO
+      spgasdev = real(rval)
+
+      end
 
-      NTOT(JSEQ) = NTOT(JSEQ) + LENV
-      IF ( NTOT(JSEQ) .GE. MODCNS )  THEN
-        NTOT2(JSEQ) = NTOT2(JSEQ) + 1
-        NTOT(JSEQ)  = NTOT(JSEQ) - MODCNS
-      ENDIF
+      SUBROUTINE RMMARD( RVEC,LENV,ISEQ )
+      implicit none
+      DOUBLE PRECISION RVEC(*), RVAL
+      INTEGER ISEQ,LENV,IVEC
+
+      DO IVEC = 1, LENV
+        call NPYRNG(RVAL)
+        RVEC(IVEC) = RVAL
```

Comment:
Are the numbers in the sequencies still randomly distirbuted in the way like this? Because before the sequencies were completly independent, and now are filled from a single sequence of random numbers. Probably it is not a problem, because it seems to me that RMMARD function isn't  used directly anywhere.